test/taskrun_test.go

```diff
@@ -136,16 +156,27 @@ spec:
 }
 
 func TestTaskRunStatus(t *testing.T) {
+	taskrunStatusTest(t, false)
+}
+
+func TestTaskRunStatusWithSpire(t *testing.T) {
+	taskrunStatusTest(t, true)
+}
+
+func taskrunStatusTest(t *testing.T, spireEnabled bool) {
 	ctx := context.Background()
 	ctx, cancel := context.WithCancel(ctx)
 	defer cancel()
 	c, namespace := setup(ctx, t)
-	t.Parallel()
```

Comment:
Because the `enable-spire` flag is changed when running with or without SPIRE. If the tests run in parallel, the config-map keeps changing while the non-spire and spire tests are running.

Reply:
I wanted to keep both testing in place. One with spire (the original) and the addition of spire checks.

Reply:
Could we run the non-spire tests first (in parallel) and then the spire tests (in parallel)? I'm not sure how much this will slow things down but it would probably be better to maintain parallelization if possible!

Reply:
Sure I can change that.

Reply:
@priyawadhwa any suggestions on how I can run the non-spire and spire tests separate? I want to avoid having multiple tests with the same code (with and without spire) and I don't think I have access to make another workflow (similar to the alpha feature flag) to sperate them out.

Reply:
The problem with extra tests (workflows) is that each prow tests uses a GCP project from the Boskos pool, and we risk to starve CI jobs if we add too many e2e jobs. 
For changes that can be accessed via the API we have a single feature flag to avoid the explosion of test combinations. For backend features though we must have dedicated flags, but we don't have a good strategy in place to test them. I would assume spire enabled for now, but we should create a path to test the disable version too.
Would you mind filing an issue to track this problem in tektoncd/pluming (or tektoncd/pipeline)

Reply:
@afrittoli Just to understand this. You are saying, for now, just use the spire enabled by default for the e2e tests and not test the disabled one. I can create an issue to track this.

Reply:
Actually, rethinking about this, would it make sense to use the alpha job to test the spire integration and to run the spire disabled tests in the regular e2e job?